File: solutions/python/difference-of-squares/1/difference_of_squares.py
def square_of_sum(count):
    return sum(n for n in range(1, count + 1))**2


def sum_of_squares(count):
    return sum(n**2 for n in range(1, count + 1))


def difference(count):
    return square_of_sum(count) - sum_of_squares(count)

File: solutions/dart/difference-of-squares/1/lib/difference_of_squares.dart
// dart 2.5.0
import 'dart:math' show pow;


class DifferenceOfSquares {
  int squareOfSum(int firstNaturalNumber){
    return List<int>.generate(firstNaturalNumber+1, (i)=>
       pow(i, 2).toInt()
    ).reduce((sum, element)=>sum+element);
  }
  int sumOfSquare(int firstNaturalNumber){
    return pow(List<int>.generate(firstNaturalNumber+1, (i)=>
      i+1
    ).reduce((sum, element)=>sum+element), 2).toInt();
  }
  int differenceOfSquares(int firstNaturalNumber){
    return squareOfSum(firstNaturalNumber) - sumOfSquare(firstNaturalNumber);
  }
}
